core/app/models/spree/credit_card.rb

```diff
@@ -66,11 +66,10 @@ def expiry=(expiry)
     #
     # @param num [String] the desired credit card number
     def number=(num)
-      @number = begin
-                  num.gsub(/[^0-9]/, '')
-                rescue
-                  nil
-                end
+      @number =
+        if num.is_a?(String)
```

Comment:
Do we know what other kinds of values do we expect here, besides nil?  E.g. `Numeric` values?  Is it worth updating the `# @param num [String]` method documentation, or deprecating unexpected value types, or calling `to_s` on everything?  (I don't want to bikeshed so I'm happy for this question to be a separate discussion from this PR)